enterprise/server/util/chunker/chunker.go

```diff
@@ -24,13 +24,22 @@ func (c *Chunker) Write(buf []byte) (int, error) {
 	return c.pw.Write(buf)
 }
 
-func (c *Chunker) Close() error {
+func (c *Chunker) Close() (bool, error) {
+	var ctxErr error
```

Comment:
I think this is still racy because the context may be cancelled after you check the context error but before you return. To avoid the race, you'd need to use a (mutex protected) variable to indicate whether the chunker has been closed already

Reply:
Done. I actually used a variable to save for the error encountered. I noticed that in the chunker's for loop, when we encounter error we close the pr with that error. If we don't call chunker.Write(..) that error won't be surfaced. And if there are errors encountered in the for loop, we don't want to proceed with Commit(..)